File: backend-api/Entities/Patients/EmergencyContact.cs
using CarePulse.Api.Entities.Base;

namespace CarePulse.Api.Entities.Patients;

public class EmergencyContact : BaseEntity
{
    public Guid PatientProfileId { get; set; }
    public PatientProfile? PatientProfile { get; set; }

    public string FullName { get; set; } = string.Empty;
    public string RelationshipToPatient { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public bool IsPrimary { get; set; }
}

File: mobile-app/lib/models/emergency_contact.dart
class EmergencyContact {
  final String? id;
  final String fullName;
  final String relationshipToPatient;
  final String phoneNumber;
  final bool isPrimary;

  EmergencyContact({
    this.id,
    required this.fullName,
    required this.relationshipToPatient,
    required this.phoneNumber,
    required this.isPrimary,
  });

  factory EmergencyContact.fromJson(Map<String, dynamic> json) => EmergencyContact(
        id: json['id'] as String?,
        fullName: json['fullName'] as String? ?? '',
        relationshipToPatient: json['relationshipToPatient'] as String? ?? '',
        phoneNumber: json['phoneNumber'] as String? ?? '',
        isPrimary: json['isPrimary'] as bool? ?? false,
      );

  Map<String, dynamic> toJson() => {
        'fullName': fullName,
        'relationshipToPatient': relationshipToPatient,
        'phoneNumber': phoneNumber,
        'isPrimary': isPrimary,
      };
}
